lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3061,13 +3079,18 @@ LogicalResult ConvertAtenOp<AtenSliceTensorOp>::matchAndRewrite(
   if (!matchPattern(op.getStart(), m_TorchConstantInt(&start)))
     return rewriter.notifyMatchFailure(op, "start must be a Scalar constant");
 
-  if (start < 0)
-    return rewriter.notifyMatchFailure(op, "Currently unsupported: start < 0");
+  if (start < 0) {
+    start = start + selfType.getShape()[dim];
+  }
 
   int64_t end;
   if (!matchPattern(op.getEnd(), m_TorchConstantInt(&end)))
     return rewriter.notifyMatchFailure(op, "end must be a Scalar constant");
 
+  if (end <= 0) {
```

Comment:
I don't think this is correct. For the `end == 0` case, `end` should remain zero. Make sure to also e2e test this edge case

Reply:
I no 0, then in the case end=0, start=-1, it will bug. The end < start will comeout. But when end=0, start=-1, it tries to get the last value.

Reply:
I tried to delete it but:
```
Trying to match "(anonymous namespace)::ConvertAtenOp<mlir::torch::Torch::AtenSliceTensorOp>"
    ** Failure : Currently unsupported: end < start
"(anonymous namespace)::ConvertAtenOp<mlir::torch::Torch::AtenSliceTensorOp>" result 0
  } -> FAILURE : pattern failed to match
} -> FAILURE : no matched legalization pattern
```